src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,198 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        this.table = (Node<K,V>[])new MyHashMap.Node[DEFAULT_INITIAL_CAPACITY];
+        this.threshold = (int) (table.length * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size > threshold) {
+            resize();
+        }
+        if (key == null) {
+            whenKeyNull(value);
+        } else if (nodeBusyAndKeysEquals(key)) {
+            table[bucketIndex(key)].value = value;
+        } else if (nodeBusyAndKeysDifferent(key)) {
+            if (noCollision(key)) {
+                table[bucketIndex(key)].next = addNode(key,value);
+                size++;
+            } else {
+                putWithCollision(key,value);
+            }
+        } else {
+            table[bucketIndex(key)] = addNode(key,value);
+            size++;
+        }
```

Comment:
1. Avoid Side Effects : better to make methods return values or use parameters rather than modifying external state
2. Use Meaningful Method Names, e.g. whenKeyNull, when key null what?
3. Too much minor methods : extracting some repeating code into smaller methods is a good practice, but don't be fanatic about this (no need to extract every single piece of code into the method)
```suggestion
        if (key == null) {
            whenKeyNull(value);
        } else if (nodeBusyAndKeysEquals(key)) {
            table[bucketIndex(key)].value = value;
        } else if (nodeBusyAndKeysDifferent(key)) {
            if (noCollision(key)) {
                table[bucketIndex(key)].next = addNode(key,value);
                size++;
            } else {
                putWithCollision(key,value);
            }
        } else {
            table[bucketIndex(key)] = addNode(key,value);
            size++;
        }
```